core/src/main/java/be/isach/ultracosmetics/player/profile/CosmeticsProfile.java

```diff
@@ -166,6 +166,69 @@ public void loadToPlayer() {
         }
     }
 
+    /**
+     * Loads the profile and enables the cosmetics without writing to the cosmetics profile.
+     */
+    public void loadToPlayerWithoutSaving() {
+        if (!ultraPlayer.isOnline()) {
+            UltraCosmeticsData.get().getPlugin().getSmartLogger().write(SmartLogger.LogLevel.ERROR, "Failed to load profile for " + ultraPlayer.getUsername() + "!");
+            return;
+        }
+
+        UltraCosmetics ultraCosmetics = UltraCosmeticsData.get().getPlugin();
+
+        // Gadget
+        if (enabledGadget != null
```

Comment:
should check isEnabled before getCategory.isEnabled since Java evaluates left to right